src/main/proto/netflix/titus/agent.proto

```diff
@@ -321,6 +321,29 @@ message TaskIdentity {
     optional string ipv6Address = 5;
 }
 
+// For the identity client to query the metadata server to confirm the
+// currently running task's identity based on the whole pod
+message TaskPodIdentity {
+    // (Required) Current state of the task 
+    optional messages.TaskInfo taskInfo = 1;
+
+    // (Required) Pod details in bytes!
+    // > For the same reason, we do not recommend embedding these proto objects within your own proto definitions.
+    // > It is better to store Kubernetes objects as byte arrays, in the wire format, which is self-describing.
+    // > This permits you to use either JSON or binary (proto) wire formats without code changes.
+    // > It will be difficult for you to operate on both Custom Resources and built-in types otherwise.
```

Comment:
Ok, so this will be the Kubernetes-proprietary "wire format" serialization of one of the k8s proto objects? Can this clarify which one? [This one](https://github.com/kubernetes/api/blob/9f22d12656517f7fc39cb292d16cb8079b5cd345/core/v1/generated.proto#L2949)?

For reference, a java library implementation for parsing k8s messages off the wire format is here: https://github.com/kubernetes-client/java/blob/f20788272291c0e79a8c831d8d5a7dd94d96d2de/util/src/main/java/io/kubernetes/client/ProtoClient.java

Reply:
You got it. 

Can you confirm this will "Work" in the sense that, we'll just have the hole pod here, and that will have everything you need for metatron to verify things?

I'm really worried about environment variables.

Previously, the whole cInfo was "immutable" and we never had any bugs (afaik?) with any mistmatches between what the titus api had and what we gave metatron on the classic task identity input. (how could there be? Nobody would be crazy to deserialize the base64 cinfo, mutate it, then base64 encode it back!)

But in this new world, the pod object is going to be mutable to some degree, and I'm worried that by the time the titus-executor has that object in ram, and presents it to metatron over this protobuf, that it will be not the same that it sees on the far side.

Reply:
For env vars, there's an annotation on the pod that indicates which env vars are added by the system: https://github.com/Netflix/titus-kube-common/blob/master/docs/examples/complete-pod.yaml#L26

Everything left over is user env vars.  We'll need to make sure that as we're adding env vars within Titus (via a webhook or in titus-executor), that we update this annotation.